src/gwt/src/org/rstudio/studio/client/RStudio.java

```diff
@@ -155,8 +155,47 @@ public void execute()
       };
    }
    
+   private static final native void maybeDelayLoadApplication(RStudio rstudio)
+   /*-{
+   	
+      if ($wnd.qt)
+      {
+         // on the desktop main window, we may need to wait for Qt to finish
+         // initialization before loading GWT
+         if ($wnd.qt.webChannelReady)
+         {
+            // Qt is ready; load the application now
+            rstudio.@org.rstudio.studio.client.RStudio::delayLoadApplication()();
+            return;
+         }
+         
+         // Qt not yet ready; set a hook and let the Qt WebChannel
+         // initialization script call it to finish initialization
+         $wnd.delayLoadApplication = $entry(function() {
+            rstudio.@org.rstudio.studio.client.RStudio::delayLoadApplication()();
+         });
+         
+         // set a timeout and attempt load just in case something goes wrong with
+         // Qt initialization (we don't want to just leave the user with a blank
+         // window)
+         setTimeout(function() {
+            if ($wnd.delayLoadApplication) {
+               $wnd.delayLoadApplication();
+               $wnd.delayLoadApplication = null;
+            }
+         }, 2000);
+         
+         return;
+      }
+      
+      // server and satellites can load as usual
+      rstudio.@org.rstudio.studio.client.RStudio::delayLoadApplication()();
```

Comment:
nit: I think this would read better in an `else` (first time I reviewed this I missed that the if block above it returns unconditionally)